umi_tools/umi_methods.py

```diff
@@ -902,6 +1014,37 @@ def filterCellBarcode(self, cell):
 
         return cell
 
+    def filterUMIBarcode(self, umi):
+        '''Filter out umi barcodes not in the whitelist, with
+        optional umi barcode error correction'''
+
+        if umi not in self.umi_whitelist:
+            if self.umi_false_to_true_map:
```

Comment:
This seems like an aweful lot of nesting just so that you can tell the difference between "Not correctable" and "correction swtiched off".

Reply:
I can't see any way around this though. I've editted the code to hopefully make it clearer.